theano/gpuarray/extra_ops.py

```diff
@@ -456,6 +454,8 @@ def c_support_code_struct(self, node, nodename):
 @register_opt2([CumsumOp], 'fast_compile')
 def use_gpu_cumsumop(op, ctx_name, inputs, outputs):
     if inputs[0].dtype == 'float32':
+        if isinstance(inputs[0].type, GpuArrayType):
+            return
```

Comment:
I had this condition because, if x is already on the GPU (of GpuArrayType), we don't need to use `as_gpuarray_variable` on them.